### ISSUE TITLE
Deploying more than 20 repositories fails completely

### ISSUE THREAD

**Issue by xginn8** [balena-io/balena-cli#1070]
- **Cli version:** 9.8.0
- **Standalone install:** seen in both 
- **Node version:** v8.0.0
- **Npm version:** 5.0.0
- **Operating system and architecture:**
seen in multiple (Arch Linux & Amazon Linux)
---

Trying to deploy more than 20 containers in a single docker-compose file fails to upload any containers.

The JSON object that holds the target repository information is truncated, as shown in the local verbose docker logging:
```
Jan 09 01:46:07 ip-172-31-53-107.ec2.internal dockerd[23202]: time="2019-01-09T01:46:07.423813355Z" level=debug msg="Failed to check for pr    esence of layer sha256:fcf19287cca04336dd3bc2ad8e88123413cecb5324dee254999ea0f90899c7c7 (sha256:d91fe322e1690c8fb3f2b684fd85335d36a45e509b156868323    2aede6d8a5e2b) in registry2.balena-cloud.com/v2/d223b589c83af29dd00f17b7861a4ef1" error="errors:\ndenied: requested access to the resource is denie    d\nerror parsing HTTP 401 response body: unexpected end of JSON input: \"\"\n"
```
Once the object is truncated, the balena client cannot authenticate against the remote registry, which causes all subsequent layer pushes to fail. No error is reported by the client and the release is reported as "successful". Obviously none of the containers are actually pushed, and therefore the device is unable to pull them subsequently. The shell is also littered with similar warnings: `Progress error: denied: requested access to the resource is denied` due to the missing token.

On the device, I see the following errors:
```
09.01.19 16:04:58 (-0500) Failed to download image 'registry2.balena-cloud.com/v2/9708fe6df417ba0fbb86a3a2e09ac40c:latest' due to '(HTTP code 404) unexpected - manifest for registry2.balena-cloud.com/v2/9708fe6df417ba0fbb86a3a2e09ac40c:latest not found '
```

**Comment by dfunckt** [comment]
This has been fixed server-side so the limit here can be lifted: https://github.com/balena-io/balena-cli/blob/master/lib/utils/compose.coffee#L394

**Comment by xginn8** [comment]
Fixed and deployed as part of v10.16.0
